### PR TITLE
reflect: add SetBytes()

### PR DESCRIPTION
**Pull request by dgryski**

### CODE REVIEW

**Comment by aykevl** [comment]
@dgryski looks like there is a typo in the test

**Comment by dgryski** [comment]
Saw that. Will fix once my kids are in bed.

**Comment by dgryski** [comment]
PTAL

File: src/reflect/value.go [hunk 1]
```diff
@@ -980,7 +980,13 @@ func (v Value) SetString(x string) {
...
 
 func (v Value) SetBytes(x []byte) {
-	panic("unimplemented: (reflect.Value).SetBytes()")
+	v.checkAddressable()
+	if v.typecode.Kind() != Slice || v.typecode.elem().Kind() != Uint8 {
+		panic("reflect.Value.SetBytes called on not []byte")
+	}
+
+	// copy the header contents over
+	*(*sliceHeader)(v.value) = *(*sliceHeader)(unsafe.Pointer(&x))
```
Comment (aykevl) [inline src/reflect/value.go]: I think this works just as well, and seems easier to read:

```suggestion
	*(*[]byte)(v.value) = x
```

**Review by aykevl — APPROVED** [review]
Looks good to me!
I was initially not expecting `SetBytes` actually copies the slice bytes (and not the slice itself), but it also happens upstream so :shrug:

File: src/reflect/value.go (hunk 1 above)
Comment (dgryski) [reply src/reflect/value.go]: Oh yeah, of course.

**Review by aykevl — APPROVED** [review]

**Pull request merged**